### PR TITLE
Freeze requirements that are pip-installed during CI and release builds

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Let us freeze more!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,8 +16,7 @@ jobs:
       - uses: actions/setup-python@v2
         with:
           python-version: 3.8
-      - run: python -m pip install --upgrade pip
```
Comment (jl-wynen) [inline .github/workflows/ci.yml:19]: Are we guaranteed to already have a suitable version of pip?

Comment (SimonHeybrock) [reply .github/workflows/ci.yml:19]: I don't really know, but what I do now is as shown in an example of the `actions/setup-python`, so I'd hope so?

Comment (SimonHeybrock) [reply .github/workflows/ci.yml:19]: Other docs (https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python) *do* still have the `--upgrade pip` call, so I am not sure what to make of this...

Comment (SimonHeybrock) [reply .github/workflows/ci.yml:19]: I added it back in.

**Review by jl-wynen — APPROVED** [review]
Ok. Maybe it does not make a significant difference. The action might be only be a few weeks or months behind.

**Pull request merged**